protocol/x/govplus/keeper/msg_server.go

```diff
@@ -15,3 +21,40 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (k msgServer) SlashValidator(
+	goCtx context.Context,
+	msg *types.MsgSlashValidator,
+) (*types.MsgSlashValidatorResponse, error) {
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
+	consAddr, err := sdk.ConsAddressFromBech32(msg.ValidatorAddress)
+	if err != nil {
+		return nil, types.ErrValidatorAddress
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	_, err = k.stakingKeeper.Slash(
+		ctx,
+		consAddr,
+		int64(msg.InfractionHeight),
+		msg.PowerAtInfractionHeight.BigInt().Int64(),
+		msg.SlashFactor,
+	)
+	if err != nil {
+		log.ErrorLogWithError(
+			ctx,
+			"error occured when slashing validator",
+			err,
+		)
+		panic(err)
```

Comment:
Why panic here instead of returning an error? 
Since `x/gov` message execution doesn't recover from panic, wouldn't this lead to consensus failure?

Reply:
Only errors if catastrophic. I.e if we used the slashing module and it errored equivalently, it would also panic.

Reply:
It won’t panic if it’s just a bad address or something.

Reply:
doesnt slashing module just [return errors](https://github.com/cosmos/cosmos-sdk/blob/b947b02dc63db0a8011a96e217197bf4559dea92/x/slashing/keeper/keeper.go#L105-L108)?

Reply:
Wouldn't this cause a panic and halt the chain if [Slash()](https://github.com/cosmos/cosmos-sdk/blob/b947b02dc63db0a8011a96e217197bf4559dea92/x/staking/keeper/slash.go#L37) returns any kind of error? For example: negative slash factor, validator is unbonded, etc

Reply:
Actually, in 0.50 gov does recover from [panic](https://github.com/cosmos/cosmos-sdk/blob/b947b02dc63db0a8011a96e217197bf4559dea92/x/gov/abci.go#L186), so at least this shouldn't cause a chain halt

Reply:
Thanks for double checking, for some reason I remembered the slashing module panic instead. I changed this to error.